sdk/search/search-documents/src/searchClient.ts

```diff
@@ -249,7 +256,10 @@ export class SearchClient<T> {
 
     try {
       const result = await this.client.documents.searchPost(
-        fullOptions,
+        {
+          ...fullOptions,
+          searchText: searchText
```

Comment:
This **should** always work, though *technically* we're supposed to honor whatever comes back from nextPageParameters, which includes searchText, which is being overridden here with the original value. 

I think this is safe, but maybe @brjohnstmsft has feelings about it?

Reply:
But, this is a private method right. So, this should work

Reply:
Well the problem is the way paging work is the server sends us back the parameters to send for the next page... and it sends us **all** the parameters that it wants, which is the full query including searchText.

Technically it only modifies `skip` and `top` today, but we aren't supposed to rely on it.

I think this should be ok since I don't think the server would change the searchText between pages, even though it could.

Reply:
Seems unlikely we'd change `searchText` between pages, but in the long run it's better to follow the service's contract for paging. That means `nextPageParameters` takes precedence.